tests/test-validate-app-receipt.js

```diff
@@ -85,11 +89,13 @@ describe('fxpay.validateAppReceipt()', function() {
 
     var validator = new helper.ReceiptValidator();
 
-    fxpay.validateAppReceipt(function(error, productInfo) {
-      assert.instanceOf(error, fxpay.errors.InvalidReceipt);
-      assert.typeOf(productInfo, 'object');
+    fxpay.validateAppReceipt().then(function() {
+      done(Error('unexpected success'));
+    }).catch(function(reason) {
+      assert.instanceOf(reason.error, fxpay.errors.InvalidReceipt);
```

Comment:
Yes, that's what the trailing catch does: ``then(...).catch().catch(done)``. Any exception in the final catch will be passed as an argument like ``done(error)`` which will cause a test failure.